plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -735,3 +747,42 @@ func haveSame(a1, a2 []string) bool {
 	}
 	return false
 }
+
+type NodeStatus struct {
+	info NodeInfo
+}
+
+func GeneralPredicates(info NodeInfo) algorithm.FitPredicate {
+	node := &NodeStatus{
+		info: info,
+	}
+	return node.SchedulerGeneralPredicates
+}
+
+func (n *NodeStatus) SchedulerGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+	node, err := n.info.GetNodeInfo(nodeName)
+	if err != nil {
+		return false, err
+	}
+	return RunGeneralPredicates(pod, nodeName, nodeInfo, node)
+}
+
+func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo, node *api.Node) (bool, error) {
+	fit, err := PodFitsResource2(pod, nodeName, nodeInfo, node)
+	if !fit {
+		return fit, err
+	}
+	fit, err = PodFitsHost(pod, nodeName, nodeInfo)
+	if !fit {
+		return fit, err
+	}
+	fit, err = PodFitsHostPorts(pod, nodeName, nodeInfo)
+	if !fit {
+		return fit, err
+	}
+	// TODO: decide whether to keep this as part of GeneralPredicates()
```

Comment:
Yes kubelet already check nodeSelector a while back. @davidopp just raised a question related to node running as standalone mode which kubelet handles it as a special case:

https://github.com/kubernetes/kubernetes/blob/80ff0cbbda4418ab40a383e08ab6f55244219e52/pkg/kubelet/kubelet.go#L2221